tests/src/upgrade.rs

```diff
@@ -689,3 +691,329 @@ fn should_not_upgrade_with_larger_total_token_supply() {
+    let nft_contract_key_1_2_0: Key = nft_contract_hash_1_2_0.into();
+
+    let number_of_tokens_pre_migration = 3usize;
+
+    // Build of prestate before migration.
+    for _i in 0..number_of_tokens_pre_migration {
+        let mint_request = ExecuteRequestBuilder::standard(
+            *DEFAULT_ACCOUNT_ADDR,
+            MINT_SESSION_WASM,
+            runtime_args! {
+                ARG_NFT_CONTRACT_HASH => nft_contract_key_1_2_0,
+                ARG_TOKEN_OWNER => Key::Account(*DEFAULT_ACCOUNT_ADDR),
+                ARG_TOKEN_META_DATA => "",
+                ARG_COLLECTION_NAME => NFT_TEST_COLLECTION.to_string()
+            },
+        )
+        .build();
+
+        builder.exec(mint_request).expect_success().commit();
+    }
+
+    let upgrade_request = ExecuteRequestBuilder::standard(
+        *DEFAULT_ACCOUNT_ADDR,
+        NFT_CONTRACT_WASM,
+        runtime_args! {
+            ARG_NFT_CONTRACT_HASH => nft_contract_key_1_2_0,
+            ARG_COLLECTION_NAME => NFT_TEST_COLLECTION.to_string(),
+            ARG_NAMED_KEY_CONVENTION => NamedKeyConventionMode::V1_0Custom as u8,
+            ARG_ACCESS_KEY_NAME_1_0_0 => format!("{PREFIX_ACCESS_KEY_NAME}_{NFT_TEST_COLLECTION}"),
+            ARG_HASH_KEY_NAME_1_0_0 => format!("{PREFIX_HASH_KEY_NAME}_{NFT_TEST_COLLECTION}"),
+            ARG_TOTAL_TOKEN_SUPPLY => 10u64
+        },
+    )
+    .build();
+
+    builder.exec(upgrade_request).expect_success().commit();
+
+    let nft_contract_hash = support::get_nft_contract_hash(&builder);
+    let nft_contract_key: Key = nft_contract_hash.into();
+
+    let number_of_tokens_at_upgrade = support::get_stored_value_from_global_state::<u64>(
+        &builder,
+        nft_contract_key,
+        vec![NUMBER_OF_MINTED_TOKENS.to_string()],
+    )
+    .expect("must get u64 value");
+
+    assert_eq!(number_of_tokens_at_upgrade, 3);
+
+    let total_token_supply_post_upgrade = support::get_stored_value_from_global_state::<u64>(
+        &builder,
+        nft_contract_key,
+        vec![ARG_TOTAL_TOKEN_SUPPLY.to_string()],
+    )
+    .expect("must get u64 value");
+
+    assert_eq!(
+        total_token_supply_post_upgrade,
+        expected_total_token_supply_post_upgrade
+    );
+}
+
+#[test]
+fn should_safely_upgrade_from_1_0_0_to_1_2_0_to_current_version() {
+    //* starting total_token_supply 100u64
+    let expected_total_token_supply_post_upgrade = 50;
+    should_safely_upgrade_from_1_0_0_to_1_2_0_to_current_version_with_reporting_mode(
+        OwnerReverseLookupMode::NoLookUp,
+        expected_total_token_supply_post_upgrade,
+    );
+    let expected_total_token_supply_post_upgrade = 50;
+    should_safely_upgrade_from_1_0_0_to_1_2_0_to_current_version_with_reporting_mode(
+        OwnerReverseLookupMode::Complete,
+        expected_total_token_supply_post_upgrade,
+    );
+}
+
+fn should_safely_upgrade_from_1_0_0_to_current_version_with_reporting_mode(
```

Comment:
This test needs to add an assertion that the pages have been created accurately

Reply:
I mimicked the test `should_safely_upgrade_in_ordinal_identifier_mode` to test expected_page_record_width and expected_page and the migration event. But the width is 1000 and the page a vector of three nfts.